src/pages/groupComparison/FisherExactTwoSidedTestLabel.tsx

```diff
@@ -0,0 +1,86 @@
+import { action, makeObservable, observable } from 'mobx';
+import { observer } from 'mobx-react';
+import * as React from 'react';
+import MutationMapperDataStore from 'shared/components/mutationMapper/MutationMapperDataStore';
+import { ComparisonGroup } from './GroupComparisonUtils';
+import { toConditionalPrecisionWithMinimum } from 'shared/lib/FormatUtils';
+import _ from 'lodash';
+import { getTwoTailedPValue } from 'shared/lib/FisherExactTestCalculator';
+import { countUniqueMutations } from 'shared/lib/MutationUtils';
+import { formatPercentValue } from 'cbioportal-utils';
+
+interface IFisherExactTwoSidedTestLabelProps {
+    dataStore: MutationMapperDataStore;
+    groups: ComparisonGroup[];
+    groupToProfiledPatientCounts: {
+        [groupUid: string]: number;
+    };
+}
+
+export const FisherExactTwoSidedTestLabel: React.FC<IFisherExactTwoSidedTestLabelProps> = observer(
+    ({
+        dataStore,
+        groups,
+        groupToProfiledPatientCounts,
+    }: IFisherExactTwoSidedTestLabelProps) => {
+        const mutationCountForActiveGeneGroupA = countUniqueMutations(
+            _.intersection(
```

Comment:
we've had performance problems with intersect before.  look up the package "fast-array-intersect" which we use elsehwere in our project and i think will be must faster.